Civi/Payment/PropertyBag.php

```diff
@@ -258,7 +268,12 @@ protected function handleLegacyPropNames($prop, $silent = FALSE) {
     if ($newName === NULL && substr($prop, -2) === '-' . \CRM_Core_BAO_LocationType::getBilling()
       && isset(static::$propMap[substr($prop, 0, -2)])
     ) {
-      $newName = substr($prop, 0, -2);
+      $billingAddressProp = substr($prop, 0, -2);
+      $newName = static::$propMap[$billingAddressProp] ?? NULL;
+      if ($newName === TRUE) {
+        // Good, modern name.
+        return $billingAddressProp;
+      }
```

Comment:
It turns out we already have good test coverage on this - if you add the properties without this change you get errors like invalid function getBilling_state_province